overlord/devicestate/devicestate.go

```diff
@@ -590,7 +662,7 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, pathSI *pat
 	}
 	if needsInstall {
 		// which needs to be installed
-		return remodelVar.InstallWithDeviceContext(ctx, st, pathSI, ms.newSnap, revOpts, userID,
+		return remodelVar.InstallWithDeviceContext(ctx, st, ms.newSnap, newSnapID, revOpts, userID,
```

Comment:
It looks like pathSI is not needed as an argument to remodelEssentialSnapTasks anymore